src/clib/pio_darray.c

```diff
@@ -162,6 +170,7 @@ int PIOc_write_darray_multi(const int ncid, const int *vid, const int ioid,
         }
     }
 
+    /* ??? */
```

Comment:
As a performance note, netCDF-4 files do not write fill data. If you don't write a portion of a variable it is not written in the file. If you later try to read it, fill values are provided. But the fill values don't get written.

So it seems like for netCDF-4 you should never try to add fill values.